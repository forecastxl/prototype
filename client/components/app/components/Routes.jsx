import React from 'react'
import Match from 'react-router/Match'
import { Header } from '../../header'
import { Home } from '../../home'
import { LoginFormContainer, SignupFormContainer, SignupConfirm, SignupSuccess } from '../../user'

function Routes() {
  return (
    <div>
      <Header />
      <Match exactly pattern="/" component={Home} />
      <Match pattern="/login" component={LoginFormContainer} />
      <Match exactly pattern="/signup" component={SignupFormContainer} />
      <Match pattern="/signup/confirm" component={SignupConfirm} />
      <Match pattern="/signup/success" component={SignupSuccess} />
    </div>
  )
}

export default Routes
