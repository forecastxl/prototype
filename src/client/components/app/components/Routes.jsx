import React from 'react'
import { Route, Switch } from 'react-router-dom'
import { Header } from '../../header'
import { NotificationsContainer } from '../../notifications'
import Home from '../../../scenes/Home'
import ForgotPassword from '../../../scenes/ForgotPassword'
import Login from '../../../scenes/Login'
import Profile from '../../../scenes/Profile'
import Signup from '../../../scenes/Signup'
import SignupConfirmContainer from '../../../scenes/SignupConfirm'
import ResetPassword from '../../../scenes/ResetPassword'
import Missing from '../../../scenes/Missing'
import ProtectedRoute from './ProtectedRoute'

const Routes = () => (
  <div>
    <Header />
    <Switch>
      {/* Login */}
      <Route exact path="/login" component={Login} />
      <Route path="/login/forgot-password" component={ForgotPassword} />
      <Route path="/login/reset-password" component={ResetPassword} />

      {/* Signup */}
      <Route exact path="/signup" component={Signup} />
      <Route path="/signup/confirm" component={SignupConfirmContainer} />

      {/* Home */}
      <ProtectedRoute exact path="/" component={Home} />

      {/* Profile */}
      <ProtectedRoute path="/profile" component={Profile} />

      {/* 404 */}
      <Route component={Missing} />
    </Switch>
    <NotificationsContainer />
  </div>
)

export default Routes
