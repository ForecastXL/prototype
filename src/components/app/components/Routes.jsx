import React from 'react'
import Match from 'react-router/Match'
import { Header } from '../../header'
import { Login } from '../../login'

function Routes() {
  return (
    <div>
      <Header />
      <Match exactly pattern="/" component={Login} />
    </div>
  )
}

export default Routes
