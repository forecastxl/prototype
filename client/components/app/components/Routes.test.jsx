import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Routes from './Routes'

describe('<Routes />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Routes />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
