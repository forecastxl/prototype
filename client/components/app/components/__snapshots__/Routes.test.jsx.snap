exports[`<Routes /> renders correctly 1`] = `
<div>
  <Header />
  <Match
    component={[Function]}
    exactly={true}
    pattern="/" />
  <Match
    component={[Function]}
    exactly={false}
    pattern="/login" />
  <Match
    component={[Function]}
    exactly={true}
    pattern="/signup" />
  <Match
    component={[Function]}
    exactly={false}
    pattern="/signup/confirm" />
  <Match
    component={[Function]}
    exactly={false}
    pattern="/signup/success" />
</div>
`;
